iterator.go

```diff
@@ -0,0 +1,63 @@
+package cdb
+
+// Iterator represents an iterator over a CDB database. It is used to sequentially
+// iterate over all kv-pairs in the database.
+type Iterator struct {
+	db     *CDB
+	pos    uint32
+	endPos uint32
+	err    error
+	key    []byte
+	value  []byte
+}
+
+// Iter creates an Iterator that can be used to iterate the database.
+func (cdb *CDB) Iter() *Iterator {
+	return &Iterator{
+		db:     cdb,
+		pos:    uint32(indexSize),
+		endPos: cdb.index[0].offset,
+	}
+}
+
+// Next reads the next key-value pair and advances the iterator one record.
+// If Next returns true it successfully updated the current key/value
+// and has more records to iterate over.
+// If Next returns false it either encountered an error (can be read through Err())
+// or it has no more records to iterate over.
+func (iter *Iterator) Next() bool {
```

Comment:
Which one in the stdlib? It seems to talk about both the Next() and Scan() methods. I have not implemented a Scan() so maybe something along the lines of

"Next reads the next key/value pair and advances the iterator one record. It returns false when the scan stops, either by reaching the end of the database or an error. After **Next** returns false, the Err method will return any error that occurred during scanning."